ReactNativeClient/lib/services/rest/Api.js

```diff
@@ -448,10 +448,13 @@ class Api {
 			output.parent_id = folder.id;
 		}
 
-		if ('source_url' in requestNote) output.source_url = requestNote.source_url;
-		if ('author' in requestNote) output.author = requestNote.author;
+		if (requestNote.source_url) output.source_url = requestNote.source_url;
+		if (requestNote.author) output.author = requestNote.author;
 		if ('user_updated_time' in requestNote) output.user_updated_time = Database.formatValue(Database.TYPE_INT, requestNote.user_updated_time);
 		if ('user_created_time' in requestNote) output.user_created_time = Database.formatValue(Database.TYPE_INT, requestNote.user_created_time);
+		if ('is_todo' in requestNote && requestNote['is_todo'] == 1) {
```

Comment:
No need to check for `requestNote['is_todo'] == 1` - if the user provides a value we should set the property to that value.